pkg/ipcache/metadata.go

```diff
@@ -239,6 +256,12 @@ func (ipc *IPCache) InjectLabels(ctx context.Context, modifiedPrefixes []netip.P
 				break
 			}
 
+			// Special case: the host identity is mutable; we need to always update the SelectorCache.
+			// If the labels are unchanged, the SelectorCache will short-circuit.
+			if newID.ID == identity.ReservedIdentityHost {
+				idsToAdd[newID.ID] = newID.Labels.LabelArray()
+			}
+
```

Comment:
it's not just SelectorCacheUpdate; we elide the entire ipcache update when the comparison is true.

Given how far this needs to be backported, I'd prefer a minimal diff.

Reply:
> it's not just SelectorCacheUpdate; we elide the entire ipcache update when the comparison is true.

Do you mean the `if oldID.ID == newID.ID && prefixInfo.Source() == oldID.Source &&` comparisons? AFAICT, if `releaseIdentity` is true, then it will also skip the ipcache update, and the selectorcache update always happens if it needs to.

> Given how far this needs to be backported, I'd prefer a minimal diff.

Fair enough. I'm not opposed to keeping this minimal and submitting the refactor in a future `main` PR.